src/scripts/cfo.zig

```diff
@@ -240,6 +225,166 @@ fn run_fuzzers(
     }
 }
 
+fn run_fuzzers_prepare_tasks(shell: *Shell, gh_token: ?[]const u8) !struct {
+    working_directory: [][]const u8,
+    seed_record: []SeedRecord,
+    weight: []u32,
+} {
+    var working_directory = std.ArrayList([]const u8).init(shell.arena.allocator());
+    var seed_record = std.ArrayList(SeedRecord).init(shell.arena.allocator());
+
+    // Fuzz an independent clone of the repository, so that CFO and the fuzzer could be on
+    // different branches (to fuzz PRs and releases).
+    shell.project_root.deleteTree("working") catch {};
+
+    { // Main branch fuzzing.
+        const commit = try run_fuzzers_commit_info(shell);
+        log.info("fuzzing commit={s} timestamp={d}", .{ commit.sha, commit.timestamp });
+
+        try shell.exec("git clone https://github.com/tigerbeetle/tigerbeetle working/main", .{});
+        try shell.pushd("./working/main");
+        defer shell.popd();
+
+        try shell.exec("git switch --detach {commit}", .{ .commit = @as([]const u8, &commit.sha) });
+        shell.exec("../zig/zig build -Drelease build_fuzz", .{}) catch |err| {
+            log.warn("failed to build: {}", .{err});
+        };
+
+        for (std.enums.values(Fuzzer)) |fuzzer| {
+            try working_directory.append("./working/main");
+            try seed_record.append(.{
+                .commit_timestamp = commit.timestamp,
+                .commit_sha = commit.sha,
+                .fuzzer = fuzzer,
+                .branch = "https://github.com/tigerbeetle/tigerbeetle",
+            });
+        }
+    }
+    const task_main_count: u32 = @intCast(seed_record.items.len);
+
+    if (gh_token != null) {
+        // Any PR labeled like 'fuzz lsm_tree'
+        const GhPullRequest = struct {
+            const Label = struct {
+                id: []const u8,
+                name: []const u8,
+                description: []const u8,
+                color: []const u8,
+            };
+            number: u32,
+            labels: []Label,
+        };
+
+        const pr_list_text = try shell.exec_stdout(
+            "gh pr list --state open --json number,labels",
+            .{},
+        );
```

Comment:
TIL non-maintainers can't add labels to issues! Which is important, since we don't want to let anyone run arbitrary code on the fuzzer machines.

Reply:
>since we don't want to let anyone run arbitrary code on the fuzzer machines.

Long term, I'd actually go in the opposite direciton and change the thread model to allow arbitrary code there, and rather tightening the scope of what fuzzing machines can do.

I wouldn't worry about someone stealing our machines to mine bitcoint or sending spam --- we have just a single machine and are not a scalable target.

I do worry about someone deliberately attacking our infra, but fuzzers should not be able to do a lot of damage. Right now, they have write acccess to devhubdb, and that's it (*). So, they could wipe our historical data, and they can XSS devhub page (which is written in a very xss-friendly way), but neither seems like a big problem to me.

(*) the  asterisk here is that the machine is stateful, and who knows which secrets might be in its bash history! If we _do_ say our fuzzers are not trusted, we should make sure they are deployed in a stateless and human-less way.